### DISCUSSION TITLE
How are the keymappings for the row/column mapped?

### DISCUSSION THREAD

**Discussion by twiggshub** [cymian/mouseless#82] [General]
Todays side thing is trying to understand how to make a truly keyboard-relative position config. Subgrid works fine as in the example below, but now I'm struggling with understanding how row/column calculation on the main grid is done. @croian, this would be lovely learn from the docs or other discussions but so far I haven't gotten anywhere.

My main grid is 3x10, and the plan is to have the rows be select-able in this order: QAZWSXEDCRFVTGBYHNUJMIK,OL.PÖ- and the rows for each respective column be: 

Q + for row, and column being  QWERTYUIOP
A + for row, and column being ASDFGHJKLÖ
Z + for row, and column being  ZXCVBNM,.-

Any ideas on how to achieve this with the 3x10 main grid configuration so the layout is in landscape mode? 

Note: The subgrid keymaps are for a standard Swedish ISO keyboard

This is my config so far. 
```

always_on_subgrid_rgba:
- 0.2
- 0.9
- 0.0
- 0.1
always_show_subgrid: false
app_version: 0.2.2-preview.0
background_rgba:
- 0.0
- 0.0
- 0.0
- 0.5
classname: AppConfig
font_size_multiplier: 0.5
grid_defn:
- border_width: 0.0
  callback: build_cell_strings
  num_cells_x: 10
  num_cells_y: 3
- border_width: 1.0
  callback: build_cell_strings
  num_cells_x: 3
  num_cells_y: 10
- border_width: 0.0
  callback: apply_strings_and_subgrid_to_cells
  num_cells_x: 2
  num_cells_y: 1
grid_rgba:
- 0.2
- 0.9
- 0.0
- 0.9
hide_cursor_on_click: false
hide_location: bottom_left
keymap:
  close dialog: AltRight
  cycle click count: null
  cycle mouse action type: null
  cycle mouse button: null
  edit config: Tab
  execute mouse action: Space
  hide overlay: CommandRight tap
  hold for drag: CommandLeft
  hold for move: AltLeft
  hold for right button: CommandRight
  release hold/drag: AltRight
  repeat last mouse action: null
  show overlay: CommandLeft tap
  switch monitor left: ShiftLeft tap
  switch monitor right: ShiftRight tap
  toggle overlay: null
mouse_action_keys: "QWERTYUIOPASDFGHJKL\xD6ZXCVBNM,.-"
move_duration_ms: 100
move_real_cursor_with_virtual: false
multi_action_timeout_ms: 400
strategy: subgrid
subgrid_dims:
- 10
- 3
subgrid_font_size_multiplier: 0.75
subgrid_mouse_action_keys: "QWERTYUIOPASDFGHJKL\xD6ZXCVBNM,.-"
subgrid_rgba:
- 0.2
- 0.9
- 0.0
- 0.9
subgrid_text_rgba:
- 0.2
- 0.9
- 0.0
- 0.5
tap_threshold_ms: 200
text_rgba:
- 0.2
- 0.9
- 0.0
- 0.9
window_opacity: 0.7
```

**Comment by cymian** [comment]
hi @twiggshub, the way i've gotten this to work okay in a 16:9 aspect ratio is to put left hand keys on the top rows and right hand keys on the bottom rows of the top- and second-level grids.  I just started #85 and put the config values and screenshot as the first comment, that I think you can use as an example, but just change some of the characters.  please let me know if it helps you with or if you have more questions!

i'll update the docs with some info on this soon, and I hope to put out a tutorial soon as well!
